packages/target-starknet/src/StarknetTarget.ts

```diff
@@ -21,30 +23,48 @@ export class StarknetTarget extends TypeChainTarget {
   transformFile(file: FileDescription): Output {
     const compiled = json.parse(file.contents)
     const name = file.path.split('/').splice(-1)[0].replace('.json', '')
-    const abiByName = byName(compiled.abi)
-
-    const { generateImportStatements, requestImport } = getModuleRequestor()
-
-    const contractInterface = `
-      interface ${name} {\n 
-        ${functions(abiByName, 'default', requestImport).join('\n')}\n
-        functions: {\n
-          ${functions(abiByName, 'default', requestImport).join('\n')}\n
-        }\n
-        callStatic: {\n
-          ${functions(abiByName, 'call', requestImport).join('\n')}\n
-        }\n
-        populateTransaction: {\n
-          ${functions(abiByName, 'populate', requestImport).join('\n')}\n
-        }\n
-        estimateFee: {\n
-          ${functions(abiByName, 'estimate', requestImport).join('\n')}\n
-        }\n
-      }\n`
 
+    const {
+      structTypes,
+      // constructorArgs,
+      functions,
+      imports,
+      impoort,
+    } = transformer(compiled.abi)
+
+    const ContractInterface = impoort('starknet', 'ContractInterface')
+    // const ContractFactory = impoort('starknet', 'ContractFactory')
```

Comment:
is this needed?

Reply:
You mean `ContractFactory`? it will be commented out when starknet.js start to interpret constructor arguments in the same way it interprets method args.